tests/queries/tests.py

```diff
@@ -3725,3 +3725,69 @@ def test_ticket_24605(self):
             ).order_by('pk'),
             [i1, i2, i3], lambda x: x
         )
+
+
+class Ticket23622Tests(TestCase):
+    @skipUnlessDBFeature('can_distinct_on_fields')
+    def test_ticket_23622(self):
+        a1 = Ticket23605A()
+        a1.save()
+        a2 = Ticket23605A()
+        a2.save()
+        c1 = Ticket23605C.objects.create(field_c0=0.0)
+        Ticket23605B(
+            modela_fk=a1, field_b0=123,
+            field_b1=datetime.date(2013, 1, 6),
+            modelc_fk=c1
+        ).save()
+        Ticket23605B(
+            modela_fk=a1, field_b0=23,
+            field_b1=datetime.date(2011, 6, 6),
+            modelc_fk=c1
+        ).save()
+        Ticket23605B(
+            modela_fk=a1, field_b0=234,
+            field_b1=datetime.date(2011, 9, 2),
+            modelc_fk=c1
+        ).save()
+        Ticket23605B(
+            modela_fk=a1, field_b0=12,
+            field_b1=datetime.date(2012, 9, 15),
+            modelc_fk=c1
+        ).save()
+        Ticket23605B(
+            modela_fk=a2, field_b0=567,
+            field_b1=datetime.date(2014, 3, 1),
+            modelc_fk=c1
+        ).save()
+        Ticket23605B(
+            modela_fk=a2, field_b0=76,
+            field_b1=datetime.date(2011, 3, 3),
+            modelc_fk=c1
+        ).save()
+        Ticket23605B(
+            modela_fk=a2, field_b0=7,
+            field_b1=datetime.date(2012, 10, 20),
+            modelc_fk=c1
+        ).save()
+        Ticket23605B(
+            modela_fk=a2, field_b0=56,
+            field_b1=datetime.date(2011, 1, 27),
+            modelc_fk=c1
+        ).save()
+
+        # Make sure __pk__in and __in work the same for related fields when
```

Comment:
this could be the docstring so it's more clear upfront what the purpose of the test is?